bids2table/_bids2table.py

```diff
@@ -112,3 +116,31 @@ def load_index(
     if split_columns:
         df = flat_to_multi_columns(df, sep=sep)
     return df
+
+
+def _filter(df: pd.DataFrame, filters: Optional[Dict[str, Any]]) -> pd.DataFrame:
+    """
+    Filter a pandas DataFrame based on a dictionary of filters.
+
+    Args:
+        df: The bids2table DataFrame to filter.
+        filters: A dictionary of filters to apply to the DataFrame. Format must be
+        either a single value or a list of values. If None, does not filter.
+
+    Returns:
+        pd.DataFrame: The filtered DataFrame.
+    """
+    if filters is None:
+        return df
+
+    for key, value in filters.items():
+        if not isinstance(value, list):
+            value = [value]
+        try:
+            df = df[df["entities"][key].isin(value)]
+        except KeyError as exc_info:
+            raise exceptions.InvalidFilterError(
```

Comment:
I don't think I would catch and re-raise a custom exception. I would just let the `KeyError` be raised. Just as informative I think and less code.